exercises/circular-buffer/canonical-data.json

```diff
@@ -0,0 +1,227 @@
+{
+  "#": [
+    "In general, these circular buffers are expected to be stateful,",
+    "and each language will operate on them differently.",
+    "Tests tend to perform a series of operations, some of which expect a certain result.",
+    "As such, this common test suite can only say in abstract terms what should be done.",
+    "",
+    "Tests will contain a number of operations. The operation will be specified in the `operation` key.",
+    "Based on the operation, other keys may be present.",
+    "read: Reading from the buffer should produce the item located at `expected`",
+    "write: Writing the item located at `item` should succeed",
+    "readShouldRail: Reading from the buffer should fail.",
```

Comment:
Excuse me! I obviously meant "complain or protest strongly and persistently about", meaning the code should *rail* at its user on an attempt to perform a read.

(Fixed)